### PR TITLE
Cicd pipeline Changes

### PR DESCRIPTION
**Pull request by aayushisharma88**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: comptests/prov/run-prov-sanity-multinode-k8s.groovy [hunk 1]
```diff
@@ -13,7 +13,7 @@ pipeline {
 		stage('CODE_CHECKOUT') {
 			steps{
 			    cleanWs()
-			    checkout([$class: 'GitSCM', branches: [[name: '*/dev']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'rel_sanity_github_auto', url: 'https://github.com/Seagate/cortx-test.git']]])
+			    checkout([$class: 'GitSCM', branches: [[name: '*/CICD-Pipeline']], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'rel_sanity_github_auto', url: 'https://github.com/Seagate/cortx-test.git']]])
```
Comment (ashukakkar) [inline comptests/prov/run-prov-sanity-multinode-k8s.groovy:16]: Change the branch to main.

**Review by ashukakkar — CHANGES_REQUESTED** [review]

**Comment by pragamj** [comment]
Please attach the collection.log and execution logs

**Comment by aayushisharma88** [comment]
[Test_Cases_OP.txt](https://github.com/Seagate/cortx-test/files/8561397/Test_Cases_OP.txt)

**Comment by aayushisharma88** [comment]
[Collections.txt](https://github.com/Seagate/cortx-test/files/8561473/Collections.txt)

**Review by pragamj — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

File: commons/commands.py [hunk 2]
```diff
@@ -497,6 +497,7 @@
 K8S_CONSUL_UPDATE_CMD = 'kubectl exec -it {} -c {} -- {}'
 GET_STATS = "consul kv get -recurse stats"
 GET_BYTECOUNT = "consul kv get -recurse bytecount"
+GET_NAMESPACE = "kubectl config set-context --current --namespace=default"
```
Comment (ashukakkar) [inline commons/commands.py:500]: This is not get command, its set command. And, make it paramerterized like below:
kubectl config set-context --current --namespace={}

File: comptests/prov/test_prov_k8s_cortx.py [hunk 3]
```diff
@@ -65,6 +66,8 @@ def setup_class(cls):
                 cls.master_node_list.append(node_obj)
             else:
                 cls.worker_node_list.append(node_obj)
+        NAMESPACE = node_obj.execute_cmd(cmd=commands.GET_NAMESPACE)
+        LOGGER.info(NAMESPACE)
```
Comment (ashukakkar) [inline comptests/prov/test_prov_k8s_cortx.py:70]: This info is not required. Please remove

File: comptests/prov/test_prov_k8s_cortx.py [hunk 4]
```diff
@@ -65,6 +66,8 @@ def setup_class(cls):
                 cls.master_node_list.append(node_obj)
             else:
                 cls.worker_node_list.append(node_obj)
+        NAMESPACE = node_obj.execute_cmd(cmd=commands.GET_NAMESPACE)
```
Comment (ashukakkar) [inline comptests/prov/test_prov_k8s_cortx.py:69]: Move it to if section in master node section.

File: comptests/prov/run-prov-sanity-multinode-k8s.groovy (hunk 1 above)
Comment (seagate-sarang-sawant) [reply comptests/prov/run-prov-sanity-multinode-k8s.groovy:16]: Are you creating CICD-Pipeline branch ?

File: comptests/prov/test_prov_k8s_cortx.py [hunk 5]
```diff
@@ -28,7 +28,7 @@
 from commons import commands
 from commons.helpers.pods_helper import LogicalNode
 from commons.utils import assert_utils
-from config import CMN_CFG, PROV_CFG
+from config import CMN_CFG, PROV_CFG, PROV_TEST_CFG
```
Comment (seagate-sarang-sawant) [inline comptests/prov/test_prov_k8s_cortx.py:31]: Make this import multi line.

File: libs/ha/ha_common_libs_k8s.py [hunk 6]
```diff
@@ -778,17 +793,23 @@ def start_random_mpu(self, event, s3_data, bucket_name, object_name, file_size,
...
         :param pod_obj: Object for master node
         :param pod_list: Data pod name list to get the hctl status
+        :param dir_path : Path to repo scripts
         :return: boolean, response
         """
         LOGGER.info("Check the overall K8s cluster status.")
+        resp = None
         try:
-            resp = pod_obj.execute_cmd(common_cmd.CLSTR_STATUS_CMD.format(self.dir_path))
+            if dir_path:
```
Comment (seagate-sarang-sawant) [inline libs/ha/ha_common_libs_k8s.py:806]: cmd_path = dir_path if dir_path else self.dir_path

The full if block can be reduced to a single line.

**Comment by seagate-sarang-sawant** [comment]
@aayushisharma88   Rebase with main before committing the fixes.

File: comptests/prov/test_prov_k8s_cortx.py (hunk 5 above)
Comment (aayushisharma88) [reply comptests/prov/test_prov_k8s_cortx.py:31]: 1. As sanity is running in my local branch so only pointing to local branch. Before merging the PR changes will be made accordingly.
2. Done.

File: libs/ha/ha_common_libs_k8s.py [hunk 7]
```diff
@@ -393,29 +393,36 @@ def ha_s3_workload_operation(
...
         :return: Boolean, response
         """
         LOGGER.info("Start the cluster")
-        resp = pod_obj.execute_cmd(common_cmd.CLSTR_START_CMD.format(self.dir_path),
-                                   read_lines=True, exc=False)
-        LOGGER.debug("Cluster start response: %s", resp)
+        cmd_path = dir_path if dir_path else self.dir_path
+        resp = pod_obj.execute_cmd(common_cmd.CLSTR_START_CMD.format(cmd_path),
+                                    read_lines=True, exc=False)
+        LOGGER.info("Cluster start response: %s".format(resp))
```
Comment (ketanarlulkar) [inline libs/ha/ha_common_libs_k8s.py:407]:
```suggestion
        LOGGER.info("Cluster start response: %s", resp)
```

**Review by ketanarlulkar — APPROVED** [review]

**Review by seagate-sarang-sawant — APPROVED** [review]

**Comment by ketanarlulkar** [comment]
@aayushisharma88 please mark the fixed comments as resolved. It blocking for merger.

**Comment by aayushisharma88** [comment]
> @aayushisharma88 please mark the fixed comments as resolved. It blocking for merger.

Done.

**Pull request merged**